selfservice/strategy/oidc/token_verifier.go

```diff
@@ -0,0 +1,42 @@
+// Copyright © 2023 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package oidc
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/coreos/go-oidc"
+)
+
+func verifyToken(ctx context.Context, keySet oidc.KeySet, config *Configuration, rawIDToken, issuerURL string) (*Claims, error) {
+	tokenAudiences := append([]string{config.ClientID}, config.AdditionalIDTokenAudiences...)
+	var token *oidc.IDToken
+	var err error
+	for _, aud := range tokenAudiences {
+		verifier := oidc.NewVerifier(issuerURL, keySet, &oidc.Config{
+			ClientID: aud,
+		})
+		token, err = verifier.Verify(ctx, rawIDToken)
+		if err != nil && strings.Contains(err.Error(), "oidc: expected audience") {
```

Comment:
No `errors.Is` possible?

Reply:
Don't think so. The library returns an inline `fmt.Errorf`: https://github.com/coreos/go-oidc/blob/v3/oidc/verify.go#L277